From 34145ea7dd6d47546bb7d7f9cbe1538d57ef8801 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:04:05 -0400
Subject: [usb] r8a66597-hcd: Convert to clk_prepare/unprepare

Message-id: <1410887463-20674-174-git-send-email-dzickus@redhat.com>
Patchwork-id: 95354
O-Subject: [RHEL7 PATCH 173/591] usb: r8a66597-hcd: Convert to clk_prepare/unprepare
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 9b547a882e9ffec67bb41a4e66b4bcc0e91a2737
Author: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
Date:   Mon Oct 28 23:49:27 2013 +0100

    usb: r8a66597-hcd: Convert to clk_prepare/unprepare

    Turn clk_enable() and clk_disable() calls into clk_prepare_enable() and
    clk_disable_unprepare() to get ready for the migration to the common
    clock framework.

    Cc: linux-usb@vger.kernel.org
    Signed-off-by: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
    Acked-by: Yoshihiro Shimoda <yoshihiro.shimoda.uh@renesas.com>
    Acked-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
    Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/r8a66597-hcd.c b/drivers/usb/host/r8a66597-hcd.c
index a6fd8f5..6404c36 100644
--- a/drivers/usb/host/r8a66597-hcd.c
+++ b/drivers/usb/host/r8a66597-hcd.c
@@ -95,7 +95,7 @@ static int r8a66597_clock_enable(struct r8a66597 *r8a66597)
  int i = 0;
 
  if (r8a66597->pdata->on_chip) {
-  clk_enable(r8a66597->clk);
+  clk_prepare_enable(r8a66597->clk);
   do {
    r8a66597_write(r8a66597, SCKE, SYSCFG0);
    tmp = r8a66597_read(r8a66597, SYSCFG0);
@@ -139,7 +139,7 @@ static void r8a66597_clock_disable(struct r8a66597 *r8a66597)
  udelay(1);
 
  if (r8a66597->pdata->on_chip) {
-  clk_disable(r8a66597->clk);
+  clk_disable_unprepare(r8a66597->clk);
  } else {
   r8a66597_bclr(r8a66597, PLLC, SYSCFG0);
   r8a66597_bclr(r8a66597, XCKE, SYSCFG0);
-- 
1.7.1